core/tests/test_utils.py

```diff
@@ -2260,6 +2260,30 @@ def _create_valid_question_data(self, default_dest_state_name):
         state.interaction.default_outcome.dest = None
         return state
 
+
+GenericTestBase = AppEngineTestBase
+
+
+class LinterTestBase(GenericTestBase):
+    """Base class for linter tests."""
+
+    def setUp(self):
+        super(LinterTestBase, self).setUp()
+        self.linter_stdout = []
+
+        def mock_print(*args):
+            """Mock for python_utils.PRINT. Append the values to print to
+            linter_stdout list.
+
+            Args:
+                *args: str. Variable length argument list of values to print in
+                    the same line of output.
+            """
+            self.linter_stdout.append(
+                ' '.join(python_utils.UNICODE(arg) for arg in args))
+
+        self.print_swap = self.swap(python_utils, 'PRINT', mock_print)
+
     def assert_same_list_elements(self, phrases, stdout):
```

Comment:
Why do we have this function? 

Is it kept to check one error from the list of all errors? If yes, to check the output of one test while all tests are run you are using this to see if from the entire logs,  does the check that I'm running added its error message, right?

Reply:
Yes